### PR TITLE
Ignore verb now also shows ghost names

### PR DESCRIPTION
**Pull request by AnturK**
For when someone is spouting memes in deadchat and you don't know their key.

### CODE REVIEW

**Review by Cyberboss — APPROVED** [review]

File: code/modules/client/verbs/ooc.dm [hunk 1]
```diff
@@ -259,9 +259,17 @@ GLOBAL_VAR_INIT(normal_ooc_colour, OOC_COLOR)
...
+		ghosts += O
+	var/choices = sortKey(GLOB.clients + ghosts)
+
+	var/selection = input("Please, select a player!", "Ignore", null, null) as null|anything in choices
 	if(!selection)
 		return
+	if(ismob(selection))
+		var/mob/dead/observer/O = selection
+		selection = O.client
 	if(selection == src)
```
Comment (Cyberboss) [inline code/modules/client/verbs/ooc.dm]: This if needs updating

**Comment by Supermichael777** [comment]
Does it show ghosts to living players?

**Comment by AnturK** [comment]
Good point.

File: code/modules/client/verbs/ooc.dm [hunk 2]
```diff
@@ -259,9 +259,19 @@ GLOBAL_VAR_INIT(normal_ooc_colour, OOC_COLOR)
 	set category = "OOC"
 	set desc ="Ignore a player's messages on the OOC channel"
 
-	var/selection = input("Please, select a player!", "Ignore", null, null) as null|anything in sortKey(GLOB.clients)
+	
+	var/list/ghosts = list()
+	if(isobserver(mob))
+		for(var/mob/dead/observer/O in GLOB.dead_mob_list)
+			ghosts += O
+	var/choices = sortKey(GLOB.clients + ghosts)
```
Comment (Cyberboss) [inline code/modules/client/verbs/ooc.dm]:
```dm
var/list/choices
if(isobserver(mob))
    var/list/ghosts = list()
    ...
    choices = sortKey(GLOB.clients + ghosts)
else
    choices = sortKey(GLOB.clients)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/client/verbs/ooc.dm [hunk 3]
```diff
@@ -259,9 +259,19 @@ GLOBAL_VAR_INIT(normal_ooc_colour, OOC_COLOR)
...
-	var/selection = input("Please, select a player!", "Ignore", null, null) as null|anything in sortKey(GLOB.clients)
-	if(!selection)
+	
+	var/see_ghost_names = isobserver(mob)
+	var/list/choices = list()
+	for(var/client/C in GLOB.clients)
+		if(isobserver(C.mob) && see_ghost_names)
+			choices["[C.mob]([C])"] = C
+		else
+			choices["[C]"] = C
```
Comment (Cyberboss) [inline code/modules/client/verbs/ooc.dm]: Is stringifying necessary here?

Comment (AnturK) [reply code/modules/client/verbs/ooc.dm]: Technically no, i'll drop it in a moment.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**